aten/src/ATen/native/mkldnn/xpu/detail/QMatmul.cpp

```diff
@@ -0,0 +1,272 @@
+#include <ATen/Tensor.h>
+#include <ATen/core/Tensor.h>
+
+#include <ATen/native/mkldnn/xpu/detail/oneDNNContext.h>
+#include <ATen/native/mkldnn/xpu/detail/Attr.h>
+
+#include <oneapi/dnnl/dnnl.hpp>
+
+namespace at::native::onednn {
+
+void quantized_matmul(
+  at::Tensor  mat1, // act
+  double input_scale,
+  int64_t input_zero_point,
+  at::Tensor  mat2, // weight
+  at::Tensor& weight_scales,
+  at::Tensor& weight_zero_points,
+  at::Tensor& b_raw,
+  at::Tensor result, // output
+  double output_scale,
+  int64_t output_zero_point,
+  std::optional<c10::ScalarType> output_dtype,
+  std::optional<at::Tensor> other, // extra input for binary-post-op
+  double other_scale,
+  int64_t other_zero_point,
+  const c10::string_view& binary_post_op,
+  double binary_alpha,
+  const c10::string_view& unary_post_op,
+  torch::List<std::optional<at::Scalar>>& unary_post_op_args,
+  c10::string_view unary_post_op_algorithm){
+
+  bool m2_trans = true;
+  auto attr = Attr(output_scale, output_zero_point);
+  construct_attr_by_post_op(
+    binary_post_op,
+    binary_alpha,
+    input_scale,
+    input_zero_point,
+    unary_post_op,
+    unary_post_op_args,
+    unary_post_op_algorithm,
+    attr
+  );
+
+  size_t dims = result.dim();
+  at::Device curDevice = at::Device(at::kXPU, c10::xpu::current_device());
+  auto engine = GpuEngineManager::Instance().get_engine(curDevice);
+  auto stream = GpuStreamManager::Instance().get_stream();
+
```

Comment:
Need to add the TORCH_CHECK about dims=2 or dim=3 in the begin of this kernel, your following logic only support these two case.

Reply:
Thanks for you kindly reminding. I have added the ASSERT in `qlinear.cpp` source file. The matrix with `dim>2` should be folded at inductor level.